src/sessions.ts

```diff
@@ -123,15 +138,23 @@ class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
 
     options = options ?? {};
 
+    if (options.snapshot === true) {
+      this[kSnapshotEnabled] = true;
+      if (options.causalConsistency === true) {
+        throw new MongoDriverError(
```

Comment:
Do we want to use a more specific error here now that they are available?

Also, this requires a prose test per spec, making a note so I don't forget